### PR TITLE
Add the SDC Location picker widget to the `main` branch

### LINKED ISSUE #880: Add Location Widget (a.k.a GPS picker) to FHIR Core
What's the difference between a location widget and a Geo-widget?
From a tech perspective, they are the same thing. They pick location GPS data but from a UI perspective, they are different. Geo Widget will use the map to pick the location data. Done is FHIR Core https://github.com/opensrp/fhircore/issues/1536 

Location widget can have 2 variants

1. Show a drop-down or location tree. This helps you select the locations already saved on the app. It mostly visualizes the location hierarchy on a form. 
2. This is an input field on a form that will pick the GPS coordinates and fill in LAT & LONG input fields in the form.|

Point 2 can be a simple button that gets the (Lat, Long) when clicked or be what Ona Data did [here](https://ona.io/home/introducing-the-select-location-widget-for-ona-data/)

Opened this ticket to track the SDK issue  - https://github.com/google/android-fhir/issues/800

### PR DESCRIPTION
**Pull request by AbdulWahabMemon**
**IMPORTANT: Where possible all PRs must be linked to a Github issue**

Fixes #880

**Engineer Checklist**
- [ ] I have written **Unit tests** for any new feature(s) and edge cases for bug fixes
- [ ] I have added any strings visible on UI components to the `strings.xml` file
- [ ] I have updated the  [CHANGELOG.md](./CHANGELOG.md) file for any notable changes to the codebase
- [ ] I have run `./gradlew spotlessApply` and `./gradlew spotlessCheck` to check my code follows the project's style guide
- [ ] I have built and run the FHIRCore app to verify my change fixes the issue and/or does not break the app 
- [ ] I have checked that this PR does NOT introduce **breaking changes** that require an update to **_Content_** and/or **_Configs_**? _If it does add a sample here or a link to exactly what changes need to be made to the content._

**Code Reviewer Checklist**
- [ ] I have verified **Unit tests** have been written for any new feature(s) and edge cases
- [ ] I have verified any strings visible on UI components are in the `strings.xml` file
- [ ] I have verifed the [CHANGELOG.md](./CHANGELOG.md) file has any notable changes to the codebase
- [ ] I have verified the solution has been implemented in a configurable and generic way for reuseable components
- [ ] I have built and run the FHIRCore app to verify the change fixes the issue and/or does not break the app

### CODE REVIEW

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireActivity.kt [hunk 1]
```diff
@@ -188,11 +189,11 @@ class QuestionnaireActivity : BaseMultiLanguageActivity() {
...
-      if (launchContextResources.isNotEmpty()) {
-        questionnaireFragmentBuilder.setQuestionnaireLaunchContexts(
-          launchContextResources.map { it.json() },
-        )
-      }
+//      if (launchContextResources.isNotEmpty()) {
+//        questionnaireFragmentBuilder.setQuestionnaireLaunchContexts(
+//          launchContextResources.map { it.json() },
+//        )
+//      }
```
Comment (pld) [inline android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireActivity.kt:196]: can you explain this please? thank you

**Review by pld — APPROVED** [review]

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireActivity.kt (hunk 1 above)
Comment (AbdulWahabMemon) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireActivity.kt]: resolved, its wasn't map as per main updates required

File: android/build.gradle.kts [hunk 2]
```diff
@@ -94,6 +94,16 @@ subprojects {
...
+
+      // Maximum Jackson libraries (excluding core) version that supports Android API Level 24:
+      // https://github.com/FasterXML/jackson-databind/issues/3658
+      force("com.fasterxml.jackson.core:jackson-annotations:2.13.4")
+      force("com.fasterxml.jackson:jackson-bom:2.13.4")
+      force("com.fasterxml.jackson.core:jackson-core:2.13.4")
+      force("com.fasterxml.jackson.core:jackson-databind:2.13.4")
+      force("com.fasterxml.jackson.module:jackson-module-jaxb-annotations:2.13.4")
+      force("com.fasterxml.jackson.datatype:jackson-datatype-jsr310:2.13.4")
+      force("com.fasterxml.jackson.dataformat:jackson-dataformat-xml:2.13.4")
```
[the remark this answers is not in the document]
Comment (AbdulWahabMemon) [reply android/build.gradle.kts]: removed

File: android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireItemViewHolderFactoryMatchersProviderFactoryImpl.kt [hunk 3]
```diff
@@ -19,18 +19,27 @@ package org.smartregister.fhircore.quest.ui.questionnaire
...
+        OpenSRPQuestionnaireItemViewHolderFactoryMatchersProviderImpl
+      else -> throw NotImplementedError()
+    }
   }
 
-  object QuestionnaireItemViewHolderFactoryMatchersProviderImpl :
+  object OpenSRPQuestionnaireItemViewHolderFactoryMatchersProviderImpl :
     QuestionnaireFragment.QuestionnaireItemViewHolderFactoryMatchersProvider() {
 
     const val BARCODE_URL = "https://fhir.labs.smartregister.org/barcode-type-widget-extension"
```
[the remark this answers is not in the document]
Comment (AbdulWahabMemon) [reply android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireItemViewHolderFactoryMatchersProviderFactoryImpl.kt]: updated

**Review by ellykits — APPROVED** [review]

**Review by Rkareko — APPROVED** [review]

**Pull request merged**